### ISSUE TITLE
HEAD request ignores trailing slash

### ISSUE THREAD

**Issue by shino** [basho/riak_cs#486]
Riak CS version 1.2.2 (tag: 1.2.2)

```
$ s3cmd ls s3://test
2013-03-11 09:14         2   s3://test/1
2013-03-11 13:33        11   s3://test/hamandeggs.txt
```

HEAD reuqest for the existing object returns 200:

```
$ s3curl.pl --id local_rcs --head -- -s -x localhost:8080 'http://test.s3.amazonaws.com/1'
HTTP/1.1 200 OK
Server: MochiWeb/1.1 WebMachine/1.9.0 (someone had painted it blue)
Last-Modified: Mon, 11 Mar 2013 00:14:36 GMT
ETag: "514faf5a71b4f0f67374c388f37aa0d7"
Date: Wed, 13 Mar 2013 02:42:20 GMT
Content-Type: application/octet-stream
Content-Length: 2

```

HEAD request with a trailing slash also returns 200:

```
$ s3curl.pl --id local_rcs --head -- -s -x localhost:8080 'http://test.s3.amazonaws.com/1/'
HTTP/1.1 200 OK
Server: MochiWeb/1.1 WebMachine/1.9.0 (someone had painted it blue)
Last-Modified: Mon, 11 Mar 2013 00:14:36 GMT
ETag: "514faf5a71b4f0f67374c388f37aa0d7"
Date: Wed, 13 Mar 2013 02:42:23 GMT
Content-Type: application/octet-stream
Content-Length: 2

```

**Comment by shino** [comment]
Fixed at the current release/1.3 branch during WM resource refactoring.

```
commit 413f88d057b80704a24942191f3df9b0e4464d7c (HEAD, origin/release/1.3, release/1.3)
Merge: 4a998d0 a6f569d
Author: Scott Lystig Fritchie <slfritchie@snookles.com>
Date:   Tue Mar 12 11:43:57 2013 -0700

    Merge pull request #480 from basho/bugfix/obj-update-metadata-when-updating-value
```

Without trailing slash => 200

```
$ s3curl.pl --id local_rcs --head -- -s -x localhost:8080 'http://test.s3.amazonaws.com/1'
HTTP/1.1 200 OK
Server: Riak CS
Last-Modified: Mon, 11 Mar 2013 09:14:36 GMT
ETag: "514faf5a71b4f0f67374c388f37aa0d7"
Date: Wed, 13 Mar 2013 02:45:36 GMT
Content-Type: application/octet-stream
Content-Length: 2

```

With trailing slash => 404

```
$ s3curl.pl --id local_rcs --head -- -s -x localhost:8080 'http://test.s3.amazonaws.com/1/'
HTTP/1.1 404 Object Not Found
Server: Riak CS
Date: Wed, 13 Mar 2013 02:45:42 GMT
Content-Length: 0

```

**Comment by shino** [comment]
close #486
